src/js/hooks/outboundImport/useOutboundImportForm.js

```diff
@@ -10,38 +10,69 @@ import { useSelector } from 'react-redux';
 import fulfillmentApi from 'api/services/FulfillmentApi';
 import packingListApi from 'api/services/PackingListApi';
 import notification from 'components/Layout/notifications/notification';
+import OutboundImportStep from 'components/stock-movement-wizard/outboundImport/OutboundImportStep';
 import { STOCK_MOVEMENT_URL } from 'consts/applicationUrls';
 import NotificationType from 'consts/notificationTypes';
 import { DateFormat } from 'consts/timeFormat';
 import useOutboundImportValidation from 'hooks/outboundImport/useOutboundImportValidation';
+import useQueryParams from 'hooks/useQueryParams';
+import useSessionStorage from 'hooks/useSessionStorage';
 import useSpinner from 'hooks/useSpinner';
 import useTranslate from 'hooks/useTranslate';
 
 const useOutboundImportForm = ({ next }) => {
   const translate = useTranslate();
   const spinner = useSpinner();
+  const queryParams = useQueryParams();
 
   const { validationSchema } = useOutboundImportValidation();
   const { currentLocation } = useSelector((state) => ({
     currentLocation: state.session.currentLocation,
   }));
+  const [cachedData, setCachedData, clearCachedData] = useSessionStorage('outbound-import', {});
 
-  const getDefaultValues = () => ({
-    description: undefined,
-    origin: {
-      id: currentLocation?.id,
-      label: `${currentLocation?.name} [${currentLocation?.locationType?.description}]`,
-    },
-    destination: undefined,
-    requestedBy: undefined,
-    dateRequested: undefined,
-    dateShipped: moment(new Date()).format(DateFormat.MMM_DD_YYYY_HH_MM_SS),
-    shipmentType: undefined,
-    trackingNumber: undefined,
-    comments: undefined,
-    expectedDeliveryDate: undefined,
-    packingList: undefined,
-  });
+  const defaultValues = useMemo(() => {
+    const values = {
+      description: undefined,
+      origin: {
+        id: currentLocation?.id,
+        label: `${currentLocation?.name} [${currentLocation?.locationType?.description}]`,
+      },
+      destination: undefined,
+      requestedBy: undefined,
+      dateRequested: undefined,
+      dateShipped: moment(new Date()).format(DateFormat.MMM_DD_YYYY_HH_MM_SS),
+      shipmentType: undefined,
+      trackingNumber: undefined,
+      comments: undefined,
+      expectedDeliveryDate: undefined,
+      packingList: undefined,
+    };
+
+    /** Load cached data:
+     * If we have saved cached data in the session-storage
+     * we want to load cached data only on CONFIRM step,
+     * otherwise we clear the session-storage data since that means that user has abandoned the form
+     * and that cached data is not relevant anymore
+    * */
+    if (!_.isEmpty(cachedData)) {
+      if (OutboundImportStep.CONFIRM === queryParams?.step) {
```

Comment:
I will probably get to this below, but why do we suddenly need to store the step in the query params, when we should have access to the current step from the `useWizard` hook?

Reply:
since we are dealing with page refreshes I needed know what page are we reloading, if we were to use just state for storing current step it would be cleared after the refresh and we wouldn't know if we need to load the cached data (when we are on CONFIRM step) or not load it when we are for example on DETAILS step

Reply:
thanks for explaining, I forgot to delete this comment after I figured out below what the case is.